solana/solana-ibc/programs/solana-ibc/src/validation_context.rs

```diff
@@ -79,27 +81,40 @@ impl ibc::ValidationContext for IbcStorage<'_, '_> {
         height: &ibc::Height,
     ) -> Result<Self::AnyConsensusState> {
         let store = self.borrow();
-        let state = if height.revision_number() == 1 {
-            store.chain.consensus_state(height.revision_height().into())?
-        } else {
-            None
+        #[cfg(feature = "witness")]
+        {
+            let (_slot, fetched_timestamp, fetched_trie_root) = store
+                .private
+                .local_consensus_state
+                .iter()
+                .find(|cs| cs.0 == height.revision_height())
+                .unwrap();
```

Comment:
This should return an error rather than panicking, no? This can be handled with simple ok_or

Reply:
yes it should, solved it by propagating the error outside as u said.